uamqp/client.py

```diff
@@ -172,6 +180,39 @@ def _client_run(self, **kwargs):
         """Perform a single Connection iteration."""
         self._connection.listen(wait=self._socket_timeout)
 
+    def _close_link(self, **kwargs):
+        if self._link:
+            self._link.detach(close=True)
+            self._link = None
+
+    def _do_retryable_operation(self, operation, *args, **kwargs):
+        retry_settings = self._error_policy.configure_retries()
+        retry_active = True
+        absolute_timeout = kwargs.pop("timeout", 0)
+        while retry_active:
+            try:
+                start_time = time.time()
+                if absolute_timeout < 0:
+                    raise TimeoutError("Operation timed out.")
+                return operation(*args, timeout=absolute_timeout, **kwargs)
+            except AMQPException as exc:
+                if absolute_timeout >= 0:
+                    retry_active = self._error_policy.increment(retry_settings, exc)
+                    if not retry_active:
+                        break
+                    time.sleep(self._error_policy.get_backoff_time(retry_settings, exc))
+                    if isinstance(exc, LinkDetach):
```

Comment:
We should avoid instance checking exceptions....

Reply:
I could probably avoid this by checking the error condition 🤔

Reply:
updated to use error condition